src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -808,6 +810,31 @@ public synchronized IslandMission getIslandMission(@NotNull Island island, @NotN
         }
     }
 
+    private synchronized String getDailyIslandMission(@NotNull Island island, int index) {
+        List<String> islandMissions = IridiumSkyblock.getInstance().getDatabaseManager().getIslandMissionTableManager().getEntries(island).stream()
+                .filter(islandMission -> islandMission.getType() == Mission.MissionType.DAILY)
+                .map(IslandMission::getMissionName)
+                .distinct()
+                .collect(Collectors.toList());
+
+        if (islandMissions.size() > index) {
+            return islandMissions.get(index);
+        }
+
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        List<String> availableMissions = IridiumSkyblock.getInstance().getMissionsList().keySet().stream()
+                .filter(mission -> IridiumSkyblock.getInstance().getMissionsList().get(mission).getMissionType() == Mission.MissionType.DAILY)
+                .filter(mission -> islandMissions.stream().noneMatch(m -> m.equals(mission)))
+                .collect(Collectors.toList());
```

Comment:
Forgot an empty line
```suggestion
                .collect(Collectors.toList());
                
```